### DISCUSSION TITLE
Not able to connect to Github from Linux system

### DISCUSSION THREAD

**Discussion by shishirvish** [community/community#70452] [New to GitHub]
Dear All,

I have 2 factor authentication enabled for my id. Within Github we have feature to enable token authentication which i user to generate a token for myself.
Currently when I am trying to connect to github from a Linux server, my account is not able to connect although the credentials are correct. Please find the connection string and error below.

$ git clone https://github.com/XXXXXXXXXX.git
Cloning into 'swan-cfg-tst'...
Username for 'https://github.com': shishir
Password for 'https://shishir@github.com':
fatal: unable to access 'https://github.com/XXXXXXXXXX.git/': The requested URL returned error: 403

Could someone please suggest what might be causing the issue and how I could fix this.

Thank You

**Comment by ghost** [comment]


**Reply by shishirvish** [reply]
I tried to use any other password or 2 way authentication, and for that i receive below output. This justifies that my generated token is correct but I am not able to connect to the link.
[eunef@by-swanprd ~]$ git clone https://github.com/XXXXXXXXXXX.git
Cloning into 'swan-cfg-tst'...
Username for 'https://github.com': shishir
Password for 'https://shishir@github.com':
fatal: Authentication failed for 'https://github.com/XXXXXXXXXXX.git/'

**Comment by elderguardian** [comment]
Hey, there is an [easy and convenient way to authenticate](https://cli.github.com/manual/gh_auth_login) in the [GitHub CLI](https://cli.github.com/) which is available as a the `gh` package on most distros. 

If you don't want to install it and are using HTTP you will need a [PAT (Personal Access Token)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) which can be generated under `Settings>Developer settings>Personal access tokens`. Use it **instead of your password** when cloning and remember than **this will only work for HTTP Git operations**:
```
$ git clone https://github.com/USERNAME/REPO.git
Username: YOUR_USERNAME
Password: YOUR_PERSONAL_ACCESS_TOKEN
```

Enabling 2FA should not change anything about this. I also have it enabled, this works for me and is specified in the [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/accessing-github-using-two-factor-authentication#authenticating-on-the-command-line-using-https).

**Reply by shishirvish** [reply]
Dear Luca,

As described, I enabled 2 factor authentication for accessing Github normally. This feature is working as required. 
For accessing Github through Linux server, I created Personal Access Token which is not working for me but working for one of my collegue. We have same privileges on the server and on Github, still my connection fails and his does not.

Thank You
